src/declarePropTypesForcing.tsx

```diff
@@ -0,0 +1,12 @@
+type DeclarePropTypesForcingType = <TPropTypes>() => <
+  TProps extends {
+    [propName in keyof TPropTypes]: any
+  }
+>(
+  props: TProps,
+) => Omit<TProps, keyof TPropTypes> & TPropTypes
+
+const declarePropTypesForcing: DeclarePropTypesForcingType = () => (props) =>
```

Comment:
I renamed (compared to what I did for Sage) `declarePropTypesNonNarrowing()` -> `declarePropTypesForcing()` because "non-narrowing" sounded like "the opposite of narrowing" ie "loosening"/superset, but there doesn't have to be any relationship between the existing type vs the new type